src/wallet.js

```diff
@@ -195,13 +193,34 @@ MyWallet.makePairingCode = function (success, error) {
   }
 };
 
-MyWallet.loginFromJSON = function (stringWallet, stringExternal, magicHashHex, password) {
+MyWallet.loginFromJSON = function (stringWallet, stringExternal, magicHashHexExternal, stringLabels, magicHashHexLabels, password) {
+  assert(stringWallet, 'Wallet JSON required');
+
+  // If metadata service returned 404, do not pass in a string.
+  var externalJSON = null;
+  var labelsJSON = null;
+
+  if (stringExternal) {
+    assert(magicHashHexExternal, 'Magic hash for external required');
```

Comment:
If the user doesn't have any external data saved (and no magic hash), they still have the external "empty state", which when stringified looks something like `"{"coinify":{"auto_login":true,"trades":[]},"sfox":{"auto_login":true,"trades":[]}}"`.